File: back_pfe/src/main/java/com/app/entity/Candidat.java
package com.app.entity;

import lombok.*;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.multipart.MultipartFile;

import javax.persistence.*;
import java.io.File;


@AllArgsConstructor
@Entity
@Table(name= "CANDIDAT_TBL")
@Getter
@Setter
@ToString
public class Candidat {

    @Id
    @GeneratedValue
    private int identifiant;
    private String nom;
    private int age;
    private String sf;
    private String formation;
    private String profil;
    private int total;
    private String adresse;
    private String demande;
    private String source;
    private String nature;
    @ManyToOne
    private FileDB cv;

    public Candidat() {
    }
}

File: front_pfe/src/app/model/candidat.ts
import { CVFile } from "./cv-file";

export class Candidat {
    identifiant: number;
    nom : string;
    age : number;
    sf : string;
    formation : string;
    profil : string;
    total : number;
    adresse : string;
    demande : string;
    source : string;
    nature :string;
    cv : CVFile; 

    constructor(identifiant: number,nom : string,age : number,sf : string,formation : string,
        profil : string,total : number,adresse : string,demande : string,source : string,
        nature :string,cv : CVFile ) {

            this.identifiant = identifiant;
            this.nom = nom;
            this.age = age;
            this.sf = sf;
            this.formation = formation;
            this.profil = profil;
            this.total = total;
            this.adresse = adresse;
            this.demande = demande;
            this.source = source;
            this.nature = nature;
            this.cv = cv;
    }

}
